lib/urlwatch/jobs.py

```diff
@@ -361,14 +361,78 @@ class BrowserJob(Job):
     __kind__ = 'browser'
 
     __required__ = ('navigate',)
+    __optional__ = ('cookies', 'data', 'ssl_no_verify', 'ignore_cached', 'http_proxy', 'https_proxy',
+                    'headers', 'ignore_connection_errors', 'ignore_http_error_codes', 'encoding', 'timeout',
+                    'ignore_timeout_errors', 'ignore_too_many_redirects')
```

Comment:
Can these optional keys be shared between the URL job and the Browser job? If nothing else, move the tuple of optionals to a named global and use it in both places, but maybe even a common ancestor class is doable?

Reply:
URL also accepts the method key which Browser does not;

https://github.com/thp/urlwatch/blob/4654eb984dae6e1d7734fefcb3ae976f2d3d0843/lib/urlwatch/jobs.py#L217-L219

Reply:
In rethinking this, my initial answer was incorrect as the Browser job (navigate key) can and should accept `method `as well, so the entirety of the code handling sub-keys is identical for both URL and Browser.

But am still quite confused as to why both `kind` **and** the key containing the URL changes between URL and Browser.

A clearer way (from an end user perspective) could be to eliminate the `kind` and simply look at the key: if `url` is found, then `requests `is used; if `navigate` is found, then `requests_html` is used.  Alternatively, the name of the key pointing to an URL doesn't change between the two types of jobs (`url` always points to the URL, and `kind` effectively becomes the _method_ of extracting data from that URL).

These are architecture and UX decisions.

Reply:
> A clearer way (from an end user perspective) could be to eliminate the kind and simply look at the key: if url is found, then requests is used; if navigate is found, then requests_html is used.

Personally I feel the exact opposite. It's much clearer to me to have a `kind` parameter to indicate the job type, and other parameters to specify the details. Combining different responsibilities (indicate type & specify details) into one parameter is much more confusing.